src/controllers/controller.line.js

```diff
@@ -66,24 +66,15 @@ module.exports = DatasetController.extend({
 			// Data
 			line._children = points;
 			// Model
-			line._model = {
-				// Appearance
-				// The default behavior of lines is to break at null values, according
-				// to https://github.com/chartjs/Chart.js/issues/2435#issuecomment-216718158
-				// This option gives lines the ability to span gaps
-				spanGaps: valueOrDefault(dataset.spanGaps, options.spanGaps),
-				tension: resolve([custom.tension, dataset.lineTension, lineElementOptions.tension]),
-				backgroundColor: resolve([custom.backgroundColor, dataset.backgroundColor, lineElementOptions.backgroundColor]),
-				borderWidth: resolve([custom.borderWidth, dataset.borderWidth, lineElementOptions.borderWidth]),
-				borderColor: resolve([custom.borderColor, dataset.borderColor, lineElementOptions.borderColor]),
-				borderCapStyle: resolve([custom.borderCapStyle, dataset.borderCapStyle, lineElementOptions.borderCapStyle]),
-				borderDash: resolve([custom.borderDash, dataset.borderDash, lineElementOptions.borderDash]),
-				borderDashOffset: resolve([custom.borderDashOffset, dataset.borderDashOffset, lineElementOptions.borderDashOffset]),
-				borderJoinStyle: resolve([custom.borderJoinStyle, dataset.borderJoinStyle, lineElementOptions.borderJoinStyle]),
-				fill: resolve([custom.fill, dataset.fill, lineElementOptions.fill]),
-				steppedLine: resolve([custom.steppedLine, dataset.steppedLine, lineElementOptions.stepped]),
-				cubicInterpolationMode: resolve([custom.cubicInterpolationMode, dataset.cubicInterpolationMode, lineElementOptions.cubicInterpolationMode]),
-			};
+			line._model = model = me._resolveLineOptions(line);
+
+			// Appearance
+			// The default behavior of lines is to break at null values, according
+			// to https://github.com/chartjs/Chart.js/issues/2435#issuecomment-216718158
+			// This option gives lines the ability to span gaps
+			model.spanGaps = valueOrDefault(dataset.spanGaps, options.spanGaps);
+			model.tension = resolve([custom.tension, dataset.lineTension, lineElementOptions.tension]);
+			model.steppedLine = resolve([custom.steppedLine, dataset.steppedLine, lineElementOptions.stepped]);
```

Comment:
We should also move these 3 last `model.*` in `_resolveLineOptions` (after the loop, [as here](/chartjs/Chart.js/blob/master/src/controllers/controller.bubble.js#L167)) since it's part of the line options, then remove the `model` and `custom` variables.

Reply:
Done